pkg/certificate/castorage/k8s/k8s.go

```diff
@@ -0,0 +1,97 @@
+package k8s
+
+import (
+	"context"
+
+	"github.com/rs/zerolog/log"
+	"k8s.io/client-go/kubernetes"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/errcode"
+	"github.com/openservicemesh/osm/pkg/version"
+)
+
+// GetCertFromKubernetes is a helper function that loads a certificate from a Kubernetes secret
+// The function returns an error only if a secret is found with invalid data.
```

Comment:
nit: this comment is a bit misleading. This function also returns an error if a secret could not be retrieved

Reply:
good catch, fixed!